app/controllers/admin/comments_controller.rb

```diff
@@ -0,0 +1,135 @@
+# ============================================================================
+# Project:   ShinyCMS (Ruby version)
+# File:      app/controllers/admin/comments_controller.rb
+# Purpose:   Controller for ShinyCMS comment/discussion admin features
+#
+# Copyright: (c) 2009-2020 Denny de la Haye https://denny.me
+#
+# ShinyCMS is free software; you can redistribute it and/or
+# modify it under the terms of the GPL (version 2 or later).
+# ============================================================================
+class Admin::CommentsController < AdminController
+  include MainSiteHelper
+
+  before_action :stash_comment, except: %i[ index update ]
+
+  # Display spam comment moderation page
+  def index
+    page_num = params[ :page ] || 1
+    @comments = Comment.all_spam.page( page_num )
+    authorise Comment
+    authorise @comments if @comments.present?
+  end
+
+  # Process submission of spam comment moderation page
+  def update
+    authorise Comment
+    the_params = update_params
+    if the_params[ :spam_or_ham ] == 'spam'
+      process_spam_comments
+    elsif the_params[ :spam_or_ham ] == 'ham'
+      process_ham_comments
+    else
+      flash[ :alert ] = t( '.spam_or_ham' )
+    end
+
+    redirect_to action: :index
+  end
+
+  def hide
+    authorise @comment
+    @comment.hide
+    redirect_to request.referer || comment_in_context_path( @comment )
```

Comment:
Consider using `redirect_back` instead - this handles the referer check. https://api.rubyonrails.org/classes/ActionController/Redirecting.html#method-i-redirect_back